internal/backend/pager/pager.go

```diff
@@ -0,0 +1,71 @@
+package pager
+
+import "fmt"
+
+const (
+	RESERVED_HEADER_PAGE_COUNT = 1
+)
+
+type file interface {
+	Read(int, int) ([]byte, error)
+	Write(int, []byte) error
+}
+
+/*
+	Pager manages the database pages for access by the BTree
+Properties
+	 - pageSize:      the number of bytes in each data page
+	 - fs:            file system operator used for reading and writing to the db
+	 - freelistIndex: the page number where the freelist starts
+
+	 The database has the following pages:
+
+	 Header:     Metadata about the database
+	 TablePages: Data pages for the database
+	 Freelist:   Unused pages to be allocated
+*/
+type Pager struct {
+	pageSize      int
+	fs            file
+	freelistIndex int
+}
+
+/*
+	Get retrieves the data page at offset i after the header
+*/
+func (p *Pager) Get(i int) ([]byte, error) {
```

Comment:
Can this have a more descriptive name?

Reply:
yeah, is a good idea - would `GetPage` suffice?